### PR TITLE
chore(portal): Split domains and set separate CAAs records

### PR DESCRIPTION
**Pull request by AndrewDryga**

### CODE REVIEW

File: terraform/environments/production/dns.tf [hunk 1]
```diff
@@ -6,25 +6,38 @@ resource "google_dns_record_set" "dns-caa" {
...
+
+resource "google_dns_record_set" "website-caa" {
+  project      = module.google-cloud-project.project.project_id
+  managed_zone = module.google-cloud-dns.zone_name
+
+  type = "CAA"
+  name = "www.${module.google-cloud-dns.dns_name}"
+  rrdatas = [
+    "0 issue \"letsencrypt.org\"",
+    "0 issue \"pki.goog\"",
```
Comment (jamilbk) [inline terraform/environments/production/dns.tf]:
```suggestion
```

We don't need Google for the `www` I think?

File: terraform/environments/production/dns.tf [hunk 2]
```diff
@@ -35,6 +48,20 @@ resource "google_dns_record_set" "website-www-redirect" {
...
 
+resource "google_dns_record_set" "blog-caa" {
+  project      = module.google-cloud-project.project.project_id
+  managed_zone = module.google-cloud-dns.zone_name
+
+  type = "CAA"
+  name = "blog.${module.google-cloud-dns.dns_name}"
+  rrdatas = [
+    "0 issue \"letsencrypt.org\"",
+    "0 issue \"pki.goog\"",
```
Comment (jamilbk) [inline terraform/environments/production/dns.tf]:
```suggestion
```

File: terraform/environments/production/dns.tf [hunk 3]
```diff
@@ -54,6 +81,20 @@ resource "google_dns_record_set" "blog-ipv6" {
...
 
+resource "google_dns_record_set" "docs-caa" {
+  project      = module.google-cloud-project.project.project_id
+  managed_zone = module.google-cloud-dns.zone_name
+
+  type = "CAA"
+  name = "docs.${module.google-cloud-dns.dns_name}"
+  rrdatas = [
+    "0 issue \"letsencrypt.org\"",
+    "0 issue \"pki.goog\"",
```
Comment (jamilbk) [inline terraform/environments/production/dns.tf]:
```suggestion
```

**Review by jamilbk — COMMENTED** [review]
Could maybe remove the goog CAA for non-google subdomains.

**Comment by AndrewDryga** [comment]
@jamilbk I've removed Google CAA from Vercel-managed website and added a few more redirects that we can do on Google Cloud. It's pretty hard to test this stuff though and deploying will be disruptive (it will take time for Google LB to provision an SSL cert and it won't do it before we update DNS records).

**Review by jamilbk — APPROVED** [review]

**Pull request merged**